hatched/hatched.py

```diff
@@ -183,86 +184,93 @@ def _load_image(
 
-        extra_args = {}
         if circular:
-            extra_args["center"] = center
-            build_func = _build_circular_hatch
+            lines = [
+                _build_circular_hatch(
+                    delta_factors[i] * hatch_pitch,
+                    offset_factors[i] * hatch_pitch,
+                    w,
+                    h,
+                    center=center,
+                )
+                for i in range(len(levels))
+            ]
         else:
-            extra_args["angle"] = hatch_angle
-            build_func = _build_diagonal_hatch
             # correct offset to ensure desired distance between hatches
             if hatch_angle % 180 != 0:
                 hatch_pitch /= math.sin((hatch_angle % 180) * math.pi / 180)
 
-        light_lines = build_func(4 * hatch_pitch, 0, w, h, **extra_args)
-        dark_lines = build_func(4 * hatch_pitch, 2 * hatch_pitch, w, h, **extra_args)
-        black_lines = build_func(2 * hatch_pitch, hatch_pitch, w, h, **extra_args)
+            lines = [
+                _build_diagonal_hatch(
+                    delta_factors[i] * hatch_pitch,
+                    offset_factors[i] * hatch_pitch,
+                    w,
+                    h,
+                    angle=hatch_angle,
+                )
+                for i in range(n_levels)
+            ]
 
         frame = Polygon([(3, 3), (w - 6, 3), (w - 6, h - 6), (3, h - 6)])
 
-        light_mls = shapely.ops.linemerge(
-            MultiLineString([line for line in light_lines])
-            .difference(light_p)
-            .intersection(frame)
-        )
-        dark_mls = shapely.ops.linemerge(
-            MultiLineString([line for line in dark_lines])
-            .difference(dark_p)
-            .intersection(frame)
-        )
-        black_mls = shapely.ops.linemerge(
-            MultiLineString([line for line in black_lines])
-            .difference(black_p)
-            .intersection(frame)
-        )
+        mls_ = [
+            MultiLineString(MultiLineString(lines[i]).difference(mask[i]).intersection(frame))
+            for i in range(n_levels)
+        ]
+
+        mls = [
+            MultiLineString(shapely.ops.linemerge([i for i in mls_[j].geoms]))
+            for j in range(n_levels)
+        ]
+
     except Exception as exc:
         print(f"Error: {exc}")
 
-    return (
-        MultiLineString(
-            [ls for ls in light_mls.geoms]
-            + [ls for ls in dark_mls.geoms]
-            + [ls for ls in black_mls.geoms]
-        ),
-        black_cnt,
-        dark_cnt,
-        light_cnt,
-    )
+    all_lines = []
+    for i in range(n_levels):
+        all_lines.extend([ls for ls in mls[i].geoms])
+
+    return (MultiLineString(all_lines), *contours)
 
 
 def hatch(
     file_path: str,
     hatch_pitch: float = 5,
-    levels: Tuple[int, int, int] = (64, 128, 192),
+    levels: Union[int, Tuple[int]] = (64, 128, 192),
```

Comment:
This should read `Tuple[int, ...]` to suggest an arbitrary number of int is acceptable.

Reply:
Fixed